configure.ac

```diff
@@ -25,7 +25,7 @@ AC_CONFIG_AUX_DIR([./scripts])
 
 m4_ifdef([AC_CONFIG_MACRO_DIRS], [AC_CONFIG_MACRO_DIRS([m4])], [AC_CONFIG_MACRO_DIR([m4])])
 
-AM_INIT_AUTOMAKE([foreign])
+AM_INIT_AUTOMAKE([foreign subdir-objects])
```

Comment:
Why is this necessary?

Reply:
We link e3sm_io_driver.la at utils folder. autoreconf showing a warning message asking it to eb enabled.

Reply:
PnetCDF does the same. Did you see the same warning? See this line.
https://github.com/Parallel-NetCDF/PnetCDF/blob/8f583ffb01116ba626622809e2267d2ac980fa7c/src/utils/ncmpidiff/Makefile.am#L15